tests/integration/conftest.py

```diff
@@ -28,8 +33,39 @@ def cachi2_image() -> utils.ContainerImage:
         cachi2_repo_root = Path(__file__).parents[2]
         utils.build_image(cachi2_repo_root, tag=cachi2_image_ref)
 
-    cachi2 = utils.ContainerImage(cachi2_image_ref)
+    cachi2 = utils.Cachi2Image(cachi2_image_ref)
     if not cachi2_image_ref.startswith("localhost/"):
         cachi2.pull_image()
 
     return cachi2
+
+
+# autouse=True: It's nicer to see the pypiserver setup logs at the beginning of the test suite.
+# Otherwise, pypiserver would start once the pip tests need it and the logs would be buried between
+# test output.
+@pytest.fixture(autouse=True, scope="session")
+def local_pypiserver() -> Iterator[None]:
+    if os.getenv("CACHI2_TEST_LOCAL_PYPISERVER") != "true":
+        yield
+        return
```

Comment:
You can use `mark.skipif` on fixtures as well which will enhance readability by just that tiny bit here :).

Reply:
Tried this, but it doesn't seem to work https://github.com/pytest-dev/pytest/issues/964

Also tried `pytest.skip` instead of `yield; return`, but it looks like that also skips the tests that rely on this fixture (and together with `autouse=True`, that means skip all tests :smile: )

Reply:
> Tried this, but it doesn't seem to work

sigh, I guess I forgot to remove the existing code when I tried it which falsely led me to thinking that it works, sorry.